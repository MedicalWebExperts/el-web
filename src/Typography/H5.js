import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

const propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.oneOf(['primary', 'secondary']),
};

const defaultProps = {
  color: 'primary',
};

const styles = theme => ({
  headline: {
    fontSize: theme.fonts.baseSize,
    color: props => theme.colors[props.color],
  },
});

const H5 = ({ classes, text }) => (
  <h5 className={classes.headline}>{text}</h5>
);

H5.propTypes = propTypes;
H5.defaultProps = defaultProps;

export default injectSheet(styles)(H5);
