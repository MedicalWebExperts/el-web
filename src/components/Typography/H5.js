import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  text: PropTypes.string.isRequired,
};

const H5 = ({ text }) => (
  <h1>{text}</h1>
);

H5.propTypes = propTypes;

export default H5;
