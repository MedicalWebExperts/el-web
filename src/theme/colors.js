export default {
  default: '#000000',
  primary: '#0000ff',
  secondary: '#00b33c',
};
