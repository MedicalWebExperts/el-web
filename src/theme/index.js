import { withTheme } from 'theming';
import ThemeProvider from './provider';
import DefaultTheme from './default';

module.exports = {
  ThemeProvider,
  withTheme,
  DefaultTheme,
};
