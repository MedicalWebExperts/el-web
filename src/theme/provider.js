import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as ThemeProviderBase } from 'theming';
import DefaultTheme from './default';

const propTypes = {
  children: PropTypes.element.isRequired,
  theme: PropTypes.oneOfType([PropTypes.shape({})]),
};

const defaultProps = {
  theme: {},
};

const ThemeProvider = ({ children, theme }) => {
  // check a util funcion for deep clone/merge
  const mergedTheme = Object.keys(theme).length === 0 ? DefaultTheme : theme;

  return (
    <ThemeProviderBase theme={mergedTheme}>
      {children}
    </ThemeProviderBase>
  );
};

ThemeProvider.propTypes = propTypes;
ThemeProvider.defaultProps = defaultProps;

export default ThemeProvider;
