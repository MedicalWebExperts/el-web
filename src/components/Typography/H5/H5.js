import React from 'react';
import PropTypes from 'prop-types';

import theme from '../../../styles';

const propTypes = {
  text: PropTypes.string.isRequired,
  secondary: PropTypes.bool,
  color: PropTypes.oneOf(['primary', 'secondary', 'default']),
  styles: PropTypes.shape({}),
};

const defaultProps = {
  color: 'default',
  secondary: false,
  styles: {},
};

const H5 = ({
  text, color, secondary, styles,
}) => (
  <h5 style={styles}>
    <style jsx>
      {`
      h5 {
        color: ${theme.colors[color]}
        font-family: ${secondary ? theme.font.secondary : theme.font.primary}
        font-size: ${theme.font.size * 1.12}px
        line-height: ${theme.font.size / 10}
        margin-top: 0
        margin-bottom: ${theme.spacing * 1.6}px
      }
    `}
    </style>
    {text}
  </h5>
);

H5.propTypes = propTypes;
H5.defaultProps = defaultProps;

export default H5;
