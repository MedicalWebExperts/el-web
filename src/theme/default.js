import colors from './colors';
import typography from './components/Typography';

export default {
  font: {
    size: 16,
    primary: '"Helvetica Neue", Helvetica, Arial, sans-serif',
    secondary: 'Georgia, "Times New Roman", Times, serif',
  },
  spacing: 25,
  typography,
  colors,
};
