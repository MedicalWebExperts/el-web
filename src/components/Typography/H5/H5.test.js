import React from 'react';
import renderer from 'react-test-renderer';

import H5 from './H5';
import { ThemeProvider } from '../../../theme';

describe('Component H5 Snapshot', () => {
  it('without arguments, take the default values', () => {
    const tree = renderer
      .create(<ThemeProvider><H5 text="Heading 5" /></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('primary color of the theme', () => {
    const tree = renderer
      .create(<ThemeProvider><H5 text="Heading 5" color="primary" /></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('secondary font-family of the theme', () => {
    const tree = renderer
      .create(<ThemeProvider><H5 text="Heading 5" secondary /></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('added optional style inline prop', () => {
    const myStyles = {
      marginTop: 30,
      lineHeight: 2,
    };
    const tree = renderer
      .create(<ThemeProvider><H5 text="Heading 1" styles={myStyles} /></ThemeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
