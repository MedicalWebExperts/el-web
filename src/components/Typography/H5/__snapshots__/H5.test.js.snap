// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component H5 Snapshot added optional style inline prop 1`] = `
<h5
  className="jsx-3885586498"
  style={
    Object {
      "lineHeight": 2,
      "marginTop": 30,
    }
  }
>
  Heading 1
</h5>
`;

exports[`Component H5 Snapshot primary color of the theme 1`] = `
<h5
  className="jsx-1042296840"
  style={Object {}}
>
  Heading 5
</h5>
`;

exports[`Component H5 Snapshot secondary font-family of the theme 1`] = `
<h5
  className="jsx-787249375"
  style={Object {}}
>
  Heading 5
</h5>
`;

exports[`Component H5 Snapshot without arguments, take the default values 1`] = `
<h5
  className="jsx-3885586498"
  style={Object {}}
>
  Heading 5
</h5>
`;
